src/app/qgshandlebadlayers.cpp

```diff
@@ -509,3 +496,154 @@ int QgsHandleBadLayers::layerCount()
 {
   return mLayerList->rowCount();
 }
+
+QString QgsHandleBadLayers::findFile( const QString &fileName, const QString &basePath, int maxDepth, int driveMargin )
+{
+  int depth = 0;
+  QDir folder = QDir( QDir( basePath ).absolutePath() );
+  QString existingBase;
+  // find the nearest existing folder
+  while ( !folder.exists() && folder.absolutePath().count( '/' ) > driveMargin )
```

Comment:
@nyalldawson Are you now ok with this approach or was there something that you wanted me to change?

Reply:
All good!